docs/docs/docs/natchez-ce3.md

```diff
@@ -0,0 +1,65 @@
+---
+layout: docs
+title: "Natchez Cats Effect 3"
+---
+
+# Natchez Cats Effect 3
+
+`natchez-ce3` is a module containing an `IOLocal` based implementation of `Trace` as well as an `Entrypoint` wrapper 
```

Comment:
Could you please add a few words about the differences between the default `Kleisli` implementation and this, and maybe what someone should consider when choosing between the 2 options? 

Tbh, I never used `IOLocal` so I am not sure in which use case would this implementation be preferable.

Reply:
Hey @voidcontext - sorry for the slow reply.
They're more or less equivalent, I just think `IOLocal` is easier to use and understand - especially for anyone that's been exposed to the `MDC` in java or `ThreadLocal`.

I suspect there's also a performance improvement to be had in using `IOLocal` over `Kleisli` and wrote very quick little test case to verify this (IOLocal consistently took between 50% and 80% of the time of the Kleisli implementation) but I'd be reluctant to assert this based on the little test case I wrote.

The source code documentation in CE3 says:
> In some scenarios, `IOLocal` can be considered as an alternative to `cats.data.Kleisli`.


Let me know if you'd like me to add something to this effect in the documentation in this PR.

Reply:
Ahh, I learned something today, thanks 😄 ... and sorry, I meant it would be nice to mention these in the docs, someone might find it useful. On a second thought though, I'll probably approve and merge this as it is and we can still update the docs in standalone PR.